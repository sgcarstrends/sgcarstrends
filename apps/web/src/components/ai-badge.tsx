import { Chip } from "@heroui/chip";
import { Sparkles } from "lucide-react";

export const AIBadge = () => (
  <Chip color="primary" variant="bordered">
    <div className="flex items-center gap-1">
      <Sparkles className="size-4" />
      AI Analysis
    </div>
  </Chip>
);
