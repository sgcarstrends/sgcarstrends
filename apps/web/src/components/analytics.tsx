"use client";

import { usePathname } from "next/navigation";
import { useEffect } from "react";

export const Analytics = () => {
  const pathname = usePathname();

  useEffect(() => {
    const url = "/api/analytics";
    const body = JSON.stringify({ pathname, referrer: document.referrer });

    // Use fetch with keepalive option as a fallback for sendBeacon
    if (navigator.sendBeacon) {
      navigator.sendBeacon(url, body);
    } else {
      void fetch(url, { method: "POST", body, keepalive: true });
    }
  }, [pathname]);

  return null;
};
