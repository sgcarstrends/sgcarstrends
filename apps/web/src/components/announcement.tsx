"use client";

import { Alert } from "@heroui/react";
import { announcements } from "@web/config";
import { usePathname } from "next/navigation";

export const Announcement = () => {
  const pathname = usePathname();

  if (announcements.length === 0) {
    return null;
  }

  // Find the first announcement that matches the current path
  // Prioritize specific path matches over global announcements
  const matchingAnnouncement = announcements.find((announcement) => {
    if (!announcement.paths) {
      return false; // Skip global announcements in first pass
    }
    return announcement.paths.some((path) => pathname.startsWith(path));
  });

  // If no specific path match, fall back to global announcements
  const activeAnnouncement =
    matchingAnnouncement ||
    announcements.find((announcement) => !announcement.paths);

  if (!activeAnnouncement) {
    return null;
  }

  return (
    <Alert radius="none" variant="solid" title={activeAnnouncement.content} />
  );
};
