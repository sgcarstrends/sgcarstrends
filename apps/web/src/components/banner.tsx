"use client";

import useStore from "@web/app/store";

export const Banner = () => {
  const { bannerContent } = useStore();

  if (!bannerContent) {
    return null;
  }

  return (
    <div className="flex items-center justify-center border-b bg-gray-50">
      <div className="container mx-auto px-6 py-4">{bannerContent}</div>
    </div>
  );
};
