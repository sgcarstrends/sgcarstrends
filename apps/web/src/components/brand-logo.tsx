import { TrendingUp } from "lucide-react";
import { APP_ENV } from "@web/config";
import { AppEnv } from "@web/types";

export const BrandLogo = () => (
  <div className="flex items-center gap-2">
    <TrendingUp className="size-6 text-blue-600 lg:size-8" />
    <div className="flex items-center gap-2">
      <span className="font-bold lg:text-xl">
        <span className="text-black">SGCars</span>
        <span className="text-blue-600">Trends</span>
      </span>
      {APP_ENV === AppEnv.DEV && (
        <span className="rounded-full bg-amber-100 px-2 py-1 text-xs font-medium text-amber-800">
          Dev
        </span>
      )}
      {APP_ENV === AppEnv.STAGING && (
        <span className="rounded-full bg-purple-100 px-2 py-1 text-xs font-medium text-purple-800">
          Staging
        </span>
      )}
    </div>
  </div>
);
