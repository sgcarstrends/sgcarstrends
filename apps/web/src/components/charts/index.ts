export { createInsightData, defaultIcons, InsightCards } from "./insights";
export { MarketShareDonut } from "./market-share";
export { TopPerformersBar } from "./top-performers";
export { TrendAreaChart } from "./trends";
