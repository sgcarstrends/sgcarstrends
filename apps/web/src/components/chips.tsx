"use client";

import { Chip } from "@heroui/react";

export const BetaChip = () => (
  <Chip color="warning" size="sm" variant="light">
    Beta
  </Chip>
);

export const NewChip = () => (
  <Chip color="success" size="sm" variant="light">
    New
  </Chip>
);
