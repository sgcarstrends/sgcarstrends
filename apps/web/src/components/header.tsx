"use client";

import { cn } from "@heroui/react";
import { BrandLogo } from "@web/components/brand-logo";
import Link from "next/link";
import { usePathname } from "next/navigation";

const NAV_ITEMS = [
  { href: "/", label: "Dashboard" },
  { href: "/blog", label: "Blog" },
  { href: "/visitors", label: "Visitors" },
  { href: "/faq", label: "FAQ" },
];

export const Header = () => {
  const pathname = usePathname();

  const isActive = (path: string) => {
    if (path === "/") {
      return pathname === "/";
    }
    return pathname.startsWith(path);
  };

  return (
    <header className="sticky top-0 z-40 px-6 py-4">
      <div className="mx-auto flex max-w-fit items-center justify-center gap-4 rounded-full px-8 py-2 shadow-lg backdrop-blur-md">
        <Link href="/">
          <BrandLogo />
        </Link>
        <nav className="flex items-center gap-4">
          {NAV_ITEMS.map(({ href, label }) => {
            return (
              <Link
                key={href}
                href={href}
                className={cn(
                  "hover:text-primary",
                  isActive(href) &&
                    "font-semibold text-primary underline decoration-2 underline-offset-4",
                )}
              >
                {label}
              </Link>
            );
          })}
        </nav>
      </div>
    </header>
  );
};
