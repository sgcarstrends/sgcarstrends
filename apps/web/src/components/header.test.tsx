import { render } from "@testing-library/react";
import { Header } from "@web/components/header";
import { vi } from "vitest";

vi.mock("next/navigation", () => ({
  usePathname: () => "/",
}));

describe("Navbar", () => {
  it("should render correctly", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
