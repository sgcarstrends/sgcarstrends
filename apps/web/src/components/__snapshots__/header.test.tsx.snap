// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > should render correctly 1`] = `
<div>
  <header
    class="sticky top-0 z-40 px-6 py-4"
  >
    <div
      class="mx-auto flex max-w-fit items-center justify-center gap-4 rounded-full px-8 py-2 shadow-lg backdrop-blur-md"
    >
      <a
        href="/"
      >
        <svg
          class="lucide lucide-trending-up size-6 text-blue-600 lg:size-8"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="22 7 13.5 15.5 8.5 10.5 2 17"
          />
          <polyline
            points="16 7 22 7 22 13"
          />
        </svg>
      </a>
      <nav
        class="flex items-center gap-4"
      >
        <a
          class="hover:text-primary font-semibold text-primary underline decoration-2 underline-offset-4"
          href="/"
        >
          Dashboard
        </a>
        <a
          class="hover:text-primary"
          href="/blog"
        >
          Blog
        </a>
        <a
          class="hover:text-primary"
          href="/visitors"
        >
          Visitors
        </a>
        <a
          class="hover:text-primary"
          href="/faq"
        >
          FAQ
        </a>
      </nav>
    </div>
  </header>
</div>
`;
